### ISSUE TITLE
Woldmap glitches

### ISSUE THREAD

**Issue by Coethium** [MinetestForFun/server-minetestforfun#503]
Often the worldmap is fully of glitches, as on the picture : 
![mff worldmap](https://cloud.githubusercontent.com/assets/20361482/17606005/80bd06ce-601d-11e6-904c-969c0c565870.png)

This bug seems to exist only at evening/night. So i guess this is a buffer overflow (or similar). The server is rebooted at morning, the buffer/cache is filling up during the day and at random time on evening/night the buffer/cache overflows.

**Comment by Darcidride** [comment]
### Knew bug
Yes, this is a knew bug. The occuration of this bug is only when we're doing a manually reboot/merge, and yes, it's often morning or evening :) (when admin and devs are available)

This bug has been explained many times, but you can't know this, for the moment :)

### The explanation
Well, it's because of our worldmapper, **mtsatellite**.

The **mtseeder** goal is to generate tiles, until the **mtseeder** isn't done the **mtwebmapper** can't displayed all levels of tiles.

The problem is : the **mtseeder** duration is approximativly 15minutes, the server daily reboot is slow because of this, because we wait the entire mtseeder tiles before starting again all servers.

When we reboot manually the server, the mtseeder can't generate all tiles, so the worldmap is displayed in a bugged way.

### MTSatellite
To understand better the way MTSatellite works, you can have a look at the project on Bitbucket : https://bitbucket.org/s_l_teichmann/mtsatellite
You're welcome if you want to improve it, the daily reboot of our server is really slow :/

**Comment by s-l-teichmann** [comment]
Why do you run mtseeder at every reboot?
You only need to do it once.
The delta updates are done by mtwebmapper.

**Comment by Darcidride** [comment]
@s-l-teichmann 
If we don't do that, we encounter this glitch : 
![mff worldmap](https://cloud.githubusercontent.com/assets/20361482/17606005/80bd06ce-601d-11e6-904c-969c0c565870.png)

**Comment by s-l-teichmann** [comment]
To fix the glitch by re-running mtseeder should not be the solution.

Looking at the image i would guess that the colors.txt is out of sync: Some tiles are
generated with water other are not. 

This may relate to your autocolor generator. You only need to update the colors.txt when you install/remove/updates mods. Blind guess: you re-generate colors.txt every reboot, too? (This also should not be necessary).

**Comment by Darcidride** [comment]
Yes, we regenerate the color.txt at each reboot, it's a really fast operation, less than 10sec needed o recreate the file, so no need to do it manually at each mod textures modification

**Comment by s-l-teichmann** [comment]
How ever this glitch should not happen. I would search for races between the re-gen of the colors.txt and mapping itself.

**Comment by Darcidride** [comment]
Ok done, to be sure the autocolormapper isn't a problem anymore i let him 30sec to done its work, i will reduce this time after some tests and if the gitch isn't anymore here

**Comment by Darcidride** [comment]
Fixed, works like a charm! Thanks to everyone

**Comment by Gael-de-Sailly** [comment]
Happens again, and very often…
![worldmap](https://user-images.githubusercontent.com/6905002/28517435-a8b4510c-7064-11e7-89f4-f7f890c94714.png)
